projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -59,19 +62,45 @@ export class TableCdkDataSource implements DataSource<TableRow> {
 
     return this.rowsChange$.pipe(
       tap(rows => this.cacheRows(rows)),
-      tap(rows => this.loadingState.next(of(rows)))
+      tap(rows => this.cacheValues(rows)),
+      tap(rows => this.loadingStateSubject.next(of(rows)))
     );
   }
 
   public disconnect(): void {
     this.rowsChange$.complete();
-    this.loadingState.complete();
+    this.loadingStateSubject.complete();
+  }
+
+  public getValues(columnConfig: TableColumnConfig): unknown[] {
+    return this.cachedValues.has(columnConfig.field)
+      ? [...this.cachedValues.get(columnConfig.field)!].sort(sortUnknown)
+      : [];
   }
 
   private cacheRows(rows: StatefulTableRow[]): void {
     this.cachedRows = rows.map(TableCdkRowUtil.cloneRow);
   }
 
+  private cacheValues(rows: StatefulTableRow[]): void {
+    this.cachedValues.clear();
+
+    rows.forEach(row => {
+      Object.entries(row).forEach((keyValueTuple: [string, unknown]) => {
+        const key = keyValueTuple[0];
+        const value = keyValueTuple[1];
+
+        if (!this.cachedValues.has(key)) {
+          this.cachedValues.set(key, new Set());
+        }
+
+        const set = this.cachedValues.get(key);
```

Comment:
This would always be defined right due to line 94? If yes, we don't need the else part in line 99.

Reply:
True, and good catch.

The reason I have the check is because the typing allows for `set` to be undefined. I could add an assertion, but saw no harm in leaving the check.

Reply:
okay

Reply:
Seems weird to me. If we want to follow the typing, no reason for 93-95, else we should assert based on our check.

Reply:
Eh, that's fine. Need to fix the other comment now so I can clean this up.